sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxCollectionCache.java

```diff
@@ -117,19 +128,20 @@ public void refresh(String resourceAddress, Map<String, Object> properties) {
 
                     });
         }
-        return Mono.empty();
+        return Mono.just(new Utils.ValueHolder<>(null));
     }
 
-    private Mono<DocumentCollection> resolveByRidAsync(
+    private Mono<Utils.ValueHolder<DocumentCollection>> resolveByRidAsync(
             String resourceId, Map<String, Object> properties) {
 
         ResourceId resourceIdParsed = ResourceId.parse(resourceId);
         String collectionResourceId = resourceIdParsed.getDocumentCollectionId().toString();
 
-        return this.collectionInfoByIdCache.getAsync(
-                collectionResourceId,
-                null,
-                () -> this.getByRidAsync(collectionResourceId, properties));
+        Mono<DocumentCollection> async = this.collectionInfoByIdCache.getAsync(
+            collectionResourceId,
+            null,
+            () -> this.getByRidAsync(collectionResourceId, properties));
+        return async.map(Utils.ValueHolder::new);
```

Comment:
`async.map(Utils.ValueHolder::new)` seems odd.

If `async` Mono may return a null value, then by the time the follow has reached `map(Utils.ValueHolder::new);` Mono will explode.

if `async` Mono can never return a null value, then why doing `async.map(Utils.ValueHolder::new);` at all?

Reply:
Good point, but `async` Mono can never return a null value. 
The reason for transforming the the `async` value to Utils.ValueHolder is because of the fact that whoever calls this method also returns null values in some other cases. 
If we don't do the transformation here, we will have to do the transformation at the caller level, and there are multiple callers, so that will be code duplication. 
So in my opinion it is better to do the transformation at one place than everywhere else. 
The same comment applies to below comments as well.